azurerm/internal/services/frontdoor/validate/validate.go

```diff
@@ -0,0 +1,172 @@
+package validate
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/frontdoor/mgmt/2019-04-01/frontdoor"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/frontdoor/helper"
+)
+
+//Frontdoor name must begin with a letter or number, end with a letter or number and may contain only letters, numbers or hyphens.
+func FrontDoorName(i interface{}, k string) (_ []string, errors []error) {
```

Comment:
This would then become
```suggestion
func ValidateFrontDoorName(i interface{}, k string) (_ []string, errors []error) {
```

Reply:
Yep. Fixed.